### PR TITLE
ATL-24 Implement notifications for user actions

### PR DESCRIPTION
**Pull request by korol1368**
TASK [ATL-24](https://alexsds.myjetbrains.com/youtrack/issue/ATL-24)

### CODE REVIEW

File: src/app/components/completed-page/completed-page.component.ts [hunk 1]
```diff
@@ -23,5 +24,9 @@ export class CompletedPageComponent implements OnInit {
     setTimeout(() => {
       this.todoService.inComplete(id);
     }, 500);
+
+    this.snackBar.open('Task incomplete', '', {
```
Comment (alexsds) [inline src/app/components/completed-page/completed-page.component.ts:28]: please, move into setTimeout function

File: src/app/components/list-page/list-page.component.ts [hunk 2]
```diff
@@ -23,5 +24,9 @@ export class ListPageComponent implements OnInit {
     setTimeout(() => {
       this.todoService.complete(id);
     }, 500);
+
+    this.snackBar.open('Task completed', '', {
```
Comment (alexsds) [inline src/app/components/list-page/list-page.component.ts:28]: please, move into setTimeout function

File: src/app/components/task-form/task-form.component.html [hunk 3]
```diff
@@ -10,5 +10,13 @@ <h2 mat-dialog-title>What's your new task?</h2>
 </mat-dialog-content>
 <mat-dialog-actions align="end">
   <button mat-raised-button (click)="onNoClick()">Cancel</button>
-  <button mat-raised-button color="accent" [mat-dialog-close]="task" [disabled]="input.invalid">Add</button>
+  <button
```
Comment (alexsds) [inline src/app/components/task-form/task-form.component.html:13]: don't need any changes

File: src/app/components/task-form/task-form.component.ts [hunk 4]
```diff
@@ -10,11 +11,17 @@ import {AddTaskComponent} from '../add-task/add-task.component';
...
-  constructor(public dialogRef: MatDialogRef<AddTaskComponent>) {}
+  constructor(public dialogRef: MatDialogRef<AddTaskComponent>, private snackBar: MatSnackBar) {}
 
   ngOnInit(): void {}
 
   onNoClick(): void {
     this.dialogRef.close();
   }
+
+  openSnackBar(): void {
```
Comment (alexsds) [inline src/app/components/task-form/task-form.component.ts:22]: don't need any changes in form, the form doesn't know about the result

File: src/app/components/task-form/task-form.component.ts [hunk 5]
```diff
@@ -10,11 +11,17 @@ import {AddTaskComponent} from '../add-task/add-task.component';
...
+    this.snackBar.open('Task added', '', {
```
Comment (alexsds) [inline src/app/components/task-form/task-form.component.ts:23]: move task added a notification to AddTaskComponent

**Review by alexsds — CHANGES_REQUESTED** [review]
please set up default options for the snack bar, also add snackbar-success class
```
{
      provide: MAT_SNACK_BAR_DEFAULT_OPTIONS,
      useValue: {
        duration: 2500,
        panelClass: ['snackbar-success'],
      },
    },
```

File: src/app/components/completed-page/completed-page.component.ts (hunk 1 above)
Comment (korol1368) [reply src/app/components/completed-page/completed-page.component.ts]: Done

File: src/app/components/list-page/list-page.component.ts (hunk 2 above)
Comment (korol1368) [reply src/app/components/list-page/list-page.component.ts]: Done

File: src/app/components/task-form/task-form.component.html (hunk 3 above)
Comment (korol1368) [reply src/app/components/task-form/task-form.component.html]: Done

File: src/app/components/task-form/task-form.component.ts (hunk 4 above)
Comment (korol1368) [reply src/app/components/task-form/task-form.component.ts]: Done

File: src/app/components/task-form/task-form.component.ts (hunk 5 above)
Comment (korol1368) [reply src/app/components/task-form/task-form.component.ts]: Done

**Review by alexsds — APPROVED** [review]

**Pull request merged**